solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/action/IndexTreeInplaceAction.java

```diff
@@ -39,53 +41,40 @@
 public class IndexTreeInplaceAction extends UpdateTreeAction {
     private static final Logger log = LoggerFactory.getLogger(IndexTreeInplaceAction.class);
 
+    private IndexingMessageSender messageSender;
+
     @Override
     public void performAction(SolrUpdateRequest updateRequest) throws IndexingException {
         log.info("Starting inplace indexing of {}", updateRequest.getPid());
 
         super.performAction(updateRequest);
 
-        // Force commit the updates currently staged
-        solrUpdateDriver.commit();
-        // Cleanup any objects in the tree that were no updated
-
-        this.deleteStaleChildren(updateRequest);
+        deleteStaleChildren(updateRequest);
 
-        if (log.isDebugEnabled()) {
-            log.debug(String.format("Finished inplace indexing of {}.  {} objects updated in {}ms",
-                    updateRequest.getPid().getRepositoryPath(), updateRequest.getChildrenPending(),
-                    System.currentTimeMillis() - updateRequest.getTimeStarted()));
-        }
+        log.debug("Finished queueing inplace indexing of {}.",
+                updateRequest.getPid().getRepositoryPath());
     }
 
     public void deleteStaleChildren(SolrUpdateRequest updateRequest) throws IndexingException {
+        Map<String, String> params = new HashMap<>();
         try {
             long startTime = updateRequest.getTimeStarted();
             Date startDate = new Date(startTime);
-            String isoDate = DateFormatUtil.formatter.format(startDate);
-
-            StringBuilder query = new StringBuilder();
-
-            // If targeting the content root object, clean out all records
-            if (getContentRootPid().equals(updateRequest.getPid())) {
-                query.append("*:*");
-            } else {
-                // Get the path facet value for the starting point, since we need the hierarchy tier.
-                BriefObjectMetadata ancestorPathBean = getRootAncestorPath(updateRequest);
-
-                // Limit cleanup scope to root pid
-                query.append(solrSettings.getFieldName(SearchFieldKeys.ANCESTOR_PATH.name())).append(':')
-                        .append(SolrSettings.sanitize(ancestorPathBean.getPath().getSearchValue()));
-            }
-
-            // Target any children with timestamp older than start time.
-            query.append(" AND ").append(solrSettings.getFieldName(SearchFieldKeys.TIMESTAMP.name()))
-                    .append(":{* TO ").append(isoDate).append("}");
-
-            solrUpdateDriver.deleteByQuery(query.toString());
-        } catch (Exception e) {
-            throw new IndexingException("Error encountered in deleteStaleChildren for "
-                    + updateRequest.getTargetID(), e);
+            String dateString = DateTimeUtil.formatDateToUTC(startDate);
+            params.put(STALE_TIMESTAMP, dateString);
+
+            // Queue cleanup action for objects in tree that were not updated during this operation
```

Comment:
The javadoc for the class says:
```
Performs an update of an object and all of its descendants. After they have all updated, any descendants which were
not updated (thereby indicating they have been deleted or removed) will be removed from the index.
```
So its garbage cleanup in case things have gotten out of sync between applications.

This is compared to IndexTreeCleanAction, which deletes everything under the path first and then repopulates it. Less complicated, but tends to worry people when their collection disappears for a few hours/days.

Reply:
ok, that makes sense. i guess should read the javadoc more carefully :-)